includes/Updater.php

```diff
@@ -0,0 +1,195 @@
+				) {
+					return;
+				}
+				$this->clear_plugin_data();
+			},
+			10,
+			2
+		);
+	}
+
+	/**
+	 * Retrieves plugin information data from custom API endpoint.
+	 *
+	 * @param false|object|array $value  The result object or array. Default false.
+	 * @param string             $action The type of information being requested from the Plugin Installation API.
+	 * @param mixed              $args   Plugin API arguments.
+	 * @return false|object|array Updated $value, or passed-through $value on failure.
+	 */
+	private function plugin_info( $value, $action, $args ) {
+		list( $plugin_slug ) = explode( '/', plugin_basename( WEBSTORIES_PLUGIN_FILE ) );
+		if ( 'plugin_information' !== $action || $plugin_slug !== $args->slug ) {
+			return $value;
+		}
+		$data = $this->fetch_plugin_data();
+		if ( ! $data ) {
+			return $value;
+		}
+		$new_data = [
+			'slug'          => $plugin_slug,
+			'name'          => $data['name'],
+			'version'       => $data['version'],
+			'author'        => '<a href="https://opensource.google.com">Google</a>',
+			'download_link' => $data['download_url'],
+			'trunk'         => $data['download_url'],
+			'tested'        => $data['tested'],
+			'requires'      => $data['requires'],
+			'requires_php'  => $data['requires_php'],
+			'last_updated'  => $data['last_updated'],
+		];
+		if ( ! empty( $data['icons'] ) ) {
+			$new_data['icons'] = $data['icons'];
+		}
+		if ( ! empty( $data['banners'] ) ) {
+			$new_data['banners'] = $data['banners'];
+		}
+		if ( ! empty( $data['banners_rtl'] ) ) {
+			$new_data['banners_rtl'] = $data['banners_rtl'];
+		}
+		return (object) $new_data;
+	}
+	/**
+	 * Retrieves plugin update data from custom API endpoint.
+	 *
+	 * @param mixed $value Update check object.
+	 * @return object Modified update check object.
+	 */
+	private function updater_data( $value ) {
+		// Stop here if the current user does not have sufficient capabilities.
+		if ( ! current_user_can( 'update_plugins' ) ) {
+			return $value;
+		}
+		$data = $this->fetch_plugin_data();
+		if ( ! $data || ! isset( $data['download_url'] ) ) {
+			return $value;
+		}
+		list( $plugin_slug ) = explode( '/', plugin_basename( WEBSTORIES_PLUGIN_FILE ) );
+		$new_data            = [
+			'id'           => 'https://github.com/google/web-stories-wp',
+			'slug'         => $plugin_slug,
+			'plugin'       => plugin_basename( WEBSTORIES_PLUGIN_FILE ),
+			'new_version'  => $data['version'],
+			'url'          => $data['url'],
+			'package'      => $data['download_url'],
+			'tested'       => $data['tested'],
+			'requires'     => $data['requires'],
+			'requires_php' => $data['requires_php'],
+		];
+		if ( ! empty( $data['icons'] ) ) {
+			$new_data['icons'] = $data['icons'];
+		}
+		if ( ! empty( $data['banners'] ) ) {
+			$new_data['banners'] = $data['banners'];
+		}
+		if ( ! empty( $data['banners_rtl'] ) ) {
+			$new_data['banners_rtl'] = $data['banners_rtl'];
+		}
+		// Return data if Web Stories plugin is not up to date.
```

Comment:
Core not checks the PHP and WordPress versions before updating. See https://meta.trac.wordpress.org/ticket/3642

If the version numbers to not match, then should we do what core plugins api does and not return a result.

Reply:
Probably unlikely that we'll ever need this, but why not.